Modules/_elementtree.c

```diff
@@ -1853,6 +1862,8 @@ element_ass_subscr(PyObject *op, PyObject *item, PyObject *value)
         if (i == -1 && PyErr_Occurred()) {
             return -1;
         }
+        // If PyNumber_AsSsize_t() clears 'self->extra',
+        // we want element_setitem() to raise IndexError.
```

Comment:
This is not related to clearing `self->extra`. The extra could be initially empty.